nomos-da/network/core/src/replication/handler.rs

```diff
@@ -156,28 +194,39 @@ impl ConnectionHandler for ReplicationHandler {
     ) -> Poll<
         ConnectionHandlerEvent<Self::OutboundProtocol, Self::OutboundOpenInfo, Self::ToBehaviour>,
     > {
-        if let Err(error) = self.poll_pending_outgoing_messages(cx) {
-            return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
-                HandlerEventToBehaviour::OutgoingMessageError { error },
-            ));
-        }
+        match self.poll_pending_outgoing_messages(cx) {
+            Ok(Some(event)) => {
+                // bubble up event
+                return Poll::Ready(event);
+            }
```

Comment:
So, it seems that this was the key reason why it didn't work? and the [waker](https://github.com/logos-co/nomos-node/pull/685/files#diff-dbcbf9be7adee5a4301bc6baf61185f77850f8a513e696115cea43959a68f86aR222) stuff.

Reply:
yeah, some issues with polling and waker.